src/aliases.js

```diff
@@ -0,0 +1,128 @@
+const fs = require('fs');
+const writeFileAtomic = require('write-file-atomic');
+const writeFileAtomicSync = writeFileAtomic.sync;
+
+const ALIASES_FILE = {directory: './settings', fileName: './settings/aliases.json'}
+
+const defaultAliases = {
+    add: ["!add"],
+    back: ["!back"],
+    brb: ["!brb", "!lurk"],
+    clear: ["!clear"],
+    close: ["!close"],
+    current: ["!current"],
+    customcode: ["!customcode", "!customcodes"],
+    dismiss: ["!dismiss", "!skip", "!complete", "!completed"],
+    level: ["!level"],
+    list: ["!list", "!queue"],
+    modnext: ["!modnext"],
+    modrandom: ["!modrandom"],
+    next: ["!next"],
+    open: ["!open"],
+    order: ["!order"],
+    pause: ["!pause"],
+    persistence: ["!persistence"],
+    position: ["!position", "!pos"],
+    punt: ["!punt"],
+    random: ["!random"],
+    remove: ["!remove", "!leave"],
+    replace: ["!replace", "!change", "!swap"],
+    restart: ["!restart"],
+    resume: ["!resume"],
+    select: ["!select"],
+    start: ["!start"],
+    submitted: ["!submitted", "!entry", "!mylevel", "!mylvl"],
+    subnext: ["!subnext"],
+    subrandom: ["!subrandom"],
+    weightedchance: ["!weightedchance", "!odds", "!chance", "!chances"],
+    weightednext: ["!weightednext"],
+    weightedrandom: ["!weightedrandom"],
+    weightedsubnext: ["!weightedsubnext"],
+    weightedsubrandom: ["!weightedsubrandom"]
+}
+
+let aliases;
+
+const Aliases = {
+    saveAliases : () => {
+        if(!fs.existsSync(ALIASES_FILE.directory)){
+            fs.mkdirSync(ALIASES_FILE.directory);
+        }
+        writeFileAtomicSync(ALIASES_FILE.fileName, JSON.stringify(aliases));
+    },
+    loadAliases : (create = false) => {
+        if(create){
+            const defaults = JSON.stringify(defaultAliases, null, 2);
+            if(!fs.existsSync(ALIASES_FILE.directory)){
+                fs.mkdirSync(ALIASES_FILE.directory, {recursive: true});
+            }
+            writeFileAtomicSync(ALIASES_FILE.fileName, defaults);
+            aliases = defaultAliases;
+        }
+        if(!create && !fs.existsSync(ALIASES_FILE.fileName)){
+            Aliases.loadAliases(true);
+        }
+        try {
+            aliases = JSON.parse(fs.readFileSync(ALIASES_FILE.fileName, { encoding: "utf8" }));
+        } catch (err) {
+            console.warn('An error occurred when trying to load %s. %s', ALIASES_FILE.fileName, err.message);
+        }
+    },
+    addAlias : (cmd, alias) => {
+        if(Aliases.isDisabled(cmd) || !Aliases.isCommand(cmd)){
+            return false;
+        }
+        if(JSON.stringify(aliases).includes(alias)){
+            return false;
+        }
+        if(!alias.startsWith("!")){
+            aliases[cmd].push("!" + alias);
+        } else {
+            aliases[cmd].push(alias);
+        }
+        Aliases.saveAliases();
+        return true;
+    },
+    isDisabled : (cmd) => {
+        return aliases[cmd].includes("disabled");
```

Comment:
If `!addalias invalidcommand !hehe` is used, then the queue crashes here:
```
/home/nya/fluid-queue/src/aliases.js:87
        return aliases[cmd].includes("disabled");
                            ^

TypeError: Cannot read properties of undefined (reading 'includes')
    at Object.isDisabled (/home/nya/fluid-queue/src/aliases.js:87:29)
    at Object.addAlias (/home/nya/fluid-queue/src/aliases.js:72:20)
    at HandleMessage (/home/nya/fluid-queue/src/index.js:326:18)
    at client.onMessageHandler (/home/nya/fluid-queue/src/chatbot.js:66:9)
    at EventEmitter.emit (/home/nya/fluid-queue/node_modules/tmi.js/lib/events.js:88:13)
    at client.emits (/home/nya/fluid-queue/node_modules/tmi.js/lib/client.js:90:13)
    at client.handleMessage (/home/nya/fluid-queue/node_modules/tmi.js/lib/client.js:1117:13)
    at /home/nya/fluid-queue/node_modules/tmi.js/lib/client.js:1228:9
    at Array.forEach (<anonymous>)
    at client._onMessage (/home/nya/fluid-queue/node_modules/tmi.js/lib/client.js:1225:8)
```

Reply:
I think `Aliases.isDisabled(cmd) || !Aliases.isCommand(cmd)` needs to be replaced with `!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)` on line 90 and line 97.

Reply:
flipped || (or)